src/relay/pass/kind_check.cc

```diff
@@ -93,21 +98,68 @@ struct KindChecker : TypeVisitor {
     }
   }
 
+  void VisitType_(const TypeCallNode* op) override {
+    // type call func should be a global type var, args should be type
```

Comment:
@MarisaKirisame Could global type vars be their own kind, since they're never supposed to appear except inside a type call? Also, should the name reflect that they're only used to name ADTs? (As opposed to general-sounding `GlobalTypeVar`; that seems almost misleading because it's only a key into the module, rather than a var that's instantiated at some point.)

Reply:
I think so. I'd like GTV as it has a nice symmetry to GV though.